processor/src/debug.rs

```diff
@@ -0,0 +1,47 @@
+use crate::{Felt, Process};
+use vm_core::STACK_TOP_SIZE;
+
+#[derive(Debug)]
+/// VmState holds a current process state information at a specific clock cycle.
+pub struct VmState {
+    pub stack: [Felt; STACK_TOP_SIZE],
+    pub clk: usize,
+    pub fmp: Felt,
```

Comment:
I know this is the first version, but in the next version we should probably do the following:

1. `stack` should be a vector which includes both the top of the stack and anything in the overflow table at that clock cycle.
2. We should also have `memory`, which would be either a `BTreeMap` or a vector of tuples `(address, value)`.